### ISSUE TITLE
asmeconf:  fully compatible now?

### ISSUE THREAD

**Issue by FrankMittelbach** [latex3/tagging-project#1054]
The class is listed as partially compatible, but there have been several updates recently explicitly related to accessibility so it should be checked if the current status is still correct or if this is now another fully compliant document class.

**Comment by John02139** [comment]
These are the classes I maintain.  Validation has been complete and stable with demo.verapdf.org.


I have not seen the issue [David](https://github.com/davidcarlisle) mentions - David, where is this problem manifested?  And what do you suggest.

The demo file [here](https://mirrors.ctan.org/macros/latex/contrib/asmeconf/examples/asmeconf-accessible-PDF-UA-2.pdf) produces this report

<img width="513" height="258" alt="Image" src="https://github.com/user-attachments/assets/242a1ced-4624-44cf-a31d-99f1c1048c62" />

**Comment by davidcarlisle** [comment]
@John02139  verapdf follows the role map so the `Conference_information` tag is seen as `Div`  and so is allowed wherever Div is allowed.

`show-pdf-tags` can do that (`--map` option) but currently in the test suite we don't as it allows specific checks for custom elements. 
(the usual example is out bible example which has custom tags for Testament, Book, Chapter and Verse, they all role map to `Sect` so verapdf would be happy with a Book nested in a Verse, but validating before role mapping allows specific checks that these are well nested.

Currently our test files have only used standard or latex-project structure elements, so this reminds me I need a way to validate classes using a different set of declared tag names (either by extending the schema or using a configuration that uses the role map)

Then the second issue that as you have not used a NS rolemap our mapping to XML currently uses a generated namespce URL which produces a non-reproducible URI so always fails our diff based l3build checking.

If you added `role/user=asme`  then you would get a fixed namespace allocated for your custom tags, but we should probably fix the generation of the default one to be stable for the test suite.

**Comment by davidcarlisle** [comment]
The current asmeconf file is marked as BAD as the generated xml (with Div for the conference element) is invalid due to 


```
      <?MarkedContent page="1" ?> Template, B
      <Formula xmlns="http://iso.org/pdf2/ssn"
         id="ID.0077"
         title="math"
         af="tag-AFfile6.tex"
         xmlns:Layout="http://iso.org/pdf/ssn/Layout"
         Layout:Placement="Inline"
        >
        <AssociatedFile name="tag-AFfile6.tex" xmlns="">
        $\mathsurround =0pt$
        </AssociatedFile>
       <?MarkedContent page="1" ?>
      </Formula>
      <?MarkedContent page="1" ?>IBTEX
```

Which fails the UA-2 requirement (not checked by veradf, I think) that math has supplied MathML.

This isn't an issue with the class at all, just bad tagging generated from the hologo BiBTeX markup with a spurious empty math list .

I may just remove the  `\hologo{BibTeX}` from the test file at 

https://github.com/latex3/tagging-project/tree/main/tagging-status/testfiles-partial/asmeconf

as the `\hlogo` issue is not related to the class/

**Comment by davidcarlisle** [comment]
aha definitely we should remove it then. Note the test files here don't necessarily need to reflect your public example files, just a small minimal test that the file loads and makes reasonable tags, but obviously borrowing the example file was a natural place to start,

So I think it would help if you could assign a namespace with role/user-NS see our bible example at

https://github.com/latex3/tagging-project/blob/84a2134a8ead00fdc05313ada98a69f0aaa808a8/project-examples/ASV/bible.tex#L103


Then the generated xml would be stable and the l3build check testing would pass,  the xml RelaxNG schema validation would fail unless we use role mapping but that's our problem not yours:-) I need a way for test files to flag either that they need a custom schema or that they should use role mapping. probably the latter as I don't want to commit to extending the schema for every classs.

**Comment by davidcarlisle** [comment]
Thanks that fixes the user defined tags issue (at least in principle, our test suite harness needs extending to cope)

But  it still fails validation  as the xml file ends

```
    </tables>
   </Div>
  </Document>
```

the inserted `figures` and `tables`  sections at the end  have ended up inside a Div rather than at the top level (they all map to Sect so  this isn't actually a UA-2 issue), but currently the schema only allows this insert at the top level so we can check why they are being inserted in unexpected places.

I can look later but @u-fischer  or @FrankMittelbach  may spot what is happening here (I've seen this before, but I don't recal what the issue was)

**Comment by John02139** [comment]
The code including the two mentioned changes and all ancillary files is in a zip archive [here](https://www.dropbox.com/scl/fi/j1i2o96h625mhcp1l3mwt/asmeconf-1.45-UA2-draft.zip?rlkey=q45wcb8lgvlntj4emsn87vjhl&dl=0).

**Comment by davidcarlisle** [comment]
To test the schema I have for now added the role/user-ns declaration to the test file. I checked in the schema that makes it valid although the automated checks don't use that yet

**Comment by John02139** [comment]
I have not implemented UA-2 with pdftex (too many issues appear), so I set an error code for that branch.  

Just to confirm, this zipped code validates properly?

I don't know specifically what might have produced the fig-in-div that appeared in the previous pdf; perhaps it needed an additional run after I deleted a figure that included subcaptions.  The class file used for both cases was the same.

**Comment by John02139** [comment]
LOL!  I don't understand PDF attributes!

Thank you all for your help with this. I now have better code than I had last night. 

If we've covered all bases, this issue can be closed. 

- John

**Comment by FrankMittelbach** [comment]
> If I understand your remarks, I should **NOT** update the material in Github’s tagging status section (per Frank’s earlier request)?

@John02139 guess asmeconf has been taken care of as part of this issue and the resulting changes. But you also have mitthesis -- not checked what we say about that if anything, so perhaps that does also need tests and perhaps a pull request claiming that it is compatible. Basically, I was alerting you to the fact that helping with updating the table and the tests for CI is very welcome

**Comment by davidcarlisle** [comment]
@FrankMittelbach  we claim mitthesis is already compatible

https://github.com/search?q=repo%3Alatex3%2Ftagging-project++path%3A%2Ftestfiles-compatible-luatex.*%5C%2Fmitthesis%5C%2F%2F&type=code
